### ISSUE TITLE
experimental:annotation crash during formatting

### ISSUE THREAD

**Issue by shashachu** [pinterest/ktlint#868]
## Observed Behavior
```
com.pinterest.ktlint.core.RuleExecutionException: java.lang.IndexOutOfBoundsException: Wrong offset: 29566. Should be in range: [0, 29472]
	at com.pinterest.ktlint.core.KtLint$format$2.invoke(KtLint.kt:386)
	at com.pinterest.ktlint.core.KtLint$format$2.invoke(KtLint.kt:43)
	at com.pinterest.ktlint.core.KtLint$visitor$2$1.invoke(KtLint.kt:248)
	at com.pinterest.ktlint.core.KtLint$visitor$2$1.invoke(KtLint.kt:43)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:574)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:575)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:575)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:575)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:575)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:575)
	at com.pinterest.ktlint.core.KtLint.visit(KtLint.kt:575)
	at com.pinterest.ktlint.core.KtLint.access$visit(KtLint.kt:43)
	at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:246)
	at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:43)
	at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:371)
	at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:79)
	at com.pinterest.ktlint.KtlintCommandLine.process(Main.kt:331)
	at com.pinterest.ktlint.KtlintCommandLine.access$process(Main.kt:116)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$2$1.call(Main.kt:257)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$2$1.call(Main.kt:116)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IndexOutOfBoundsException: Wrong offset: 29566. Should be in range: [0, 29472]
	at org.jetbrains.kotlin.com.intellij.openapi.editor.impl.LineSet.findLineIndex(LineSet.java:178)
	at org.jetbrains.kotlin.com.intellij.openapi.editor.impl.DocumentImpl.getLineNumber(DocumentImpl.java:991)
	at com.pinterest.ktlint.core.ast.PackageKt.lineNumber(package.kt:235)
	at com.pinterest.ktlint.ruleset.experimental.AnnotationRule.visit(AnnotationRule.kt:149)
	at com.pinterest.ktlint.core.KtLint$format$2.invoke(KtLint.kt:376)
```

## Steps to Reproduce
```
package a.b.c

class Test {
    fun bloop() {
        asdfadsf(asdfadsf, asdfasdf, asdfasdfasdfads,
        asdfasdf, asdfasdf, asdfasdf)
    }

    @Blah
    val test: Int
}
```

## Your Environment
* Version of ktlint used: 0.38.1

Introduced in https://github.com/pinterest/ktlint/commit/4dda69b71fbbde261711b5424bf4846977648bac

**Comment by shashachu** [comment]
This appears to not play well with other rules (in this case parameter wrapping) that change the number of lines in the file.

**Comment by AdamMTGreenberg** [comment]
I can take this on and adapt it tonight

**Comment by AdamMTGreenberg** [comment]
Sorry for the delay - weekend got away from me, hoping to work on this tomorrow

**Comment by AdamMTGreenberg** [comment]
Was there a change in the setup? I am getting an error on build constantly 

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':buildSrc'.
> Dependency verification failed for configuration ':buildSrc:classpath'
  22 artifacts failed verification:
    - annotations-13.0.pom (org.jetbrains:annotations:13.0) from repository MavenRepo
    - gradle-download-task-4.0.2.pom (de.undercouch:gradle-download-task:4.0.2) from repository MavenRepo
    - gson-2.8.5.pom (com.google.code.gson:gson:2.8.5) from repository MavenRepo
    - gson-parent-2.8.5.pom (com.google.code.gson:gson-parent:2.8.5) from repository MavenRepo
    - kotlin-annotation-processing-gradle-1.3.72.pom (org.jetbrains.kotlin:kotlin-annotation-processing-gradle:1.3.72) from repository MavenRepo
    - kotlin-compiler-embeddable-1.3.72.pom (org.jetbrains.kotlin:kotlin-compiler-embeddable:1.3.72) from repository MavenRepo
    - kotlin-daemon-embeddable-1.3.72.pom (org.jetbrains.kotlin:kotlin-daemon-embeddable:1.3.72) from repository MavenRepo
    - kotlin-reflect-1.3.72.pom (org.jetbrains.kotlin:kotlin-reflect:1.3.72) from repository MavenRepo
    - kotlin-script-runtime-1.3.72.pom (org.jetbrains.kotlin:kotlin-script-runtime:1.3.72) from repository MavenRepo
    - kotlin-scripting-common-1.3.72.pom (org.jetbrains.kotlin:kotlin-scripting-common:1.3.72) from repository MavenRepo
    - kotlin-scripting-compiler-embeddable-1.3.72.pom (org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:1.3.72) from repository MavenRepo
    - kotlin-scripting-compiler-impl-embeddable-1.3.72.pom (org.jetbrains.kotlin:kotlin-scripting-compiler-impl-embeddable:1.3.72) from repository MavenRepo
    - kotlin-scripting-jvm-1.3.72.pom (org.jetbrains.kotlin:kotlin-scripting-jvm:1.3.72) from repository MavenRepo
    - kotlin-stdlib-1.3.72.pom (org.jetbrains.kotlin:kotlin-stdlib:1.3.72) from repository MavenRepo
    - kotlin-stdlib-common-1.3.72.pom (org.jetbrains.kotlin:kotlin-stdlib-common:1.3.72) from repository MavenRepo
    - kotlin-stdlib-jdk7-1.3.72.pom (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.72) from repository MavenRepo
    - kotlin-stdlib-jdk8-1.3.72.pom (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.72) from repository MavenRepo
    - kotlinx-coroutines-core-1.2.1.pom (org.jetbrains.kotlinx:kotlinx-coroutines-core:1.2.1) from repository MavenRepo
    - org.gradle.kotlin.kotlin-dsl.gradle.plugin-1.3.6.pom (org.gradle.kotlin.kotlin-dsl:org.gradle.kotlin.kotlin-dsl.gradle.plugin:1.3.6) from repository Gradle Central Plugin Repository
    - oss-parent-7.pom (org.sonatype.oss:oss-parent:7) from repository MavenRepo
    - plugins-1.3.6.pom (org.gradle.kotlin:plugins:1.3.6) from repository Gradle Central Plugin Repository
    - trove4j-1.0.20181211.pom (org.jetbrains.intellij.deps:trove4j:1.0.20181211) from repository MavenRepo
  This can indicate that a dependency has been compromised. Please carefully verify the checksums.
```

**Comment by AdamMTGreenberg** [comment]
@shashachu when I add the base test provided, it does not seem to fail - please see: https://github.com/pinterest/ktlint/pull/880

I think the underlying issue is the code above as you pointed out `AnnotationRule.visit` has me a bit troubled and I think I'm trying to do too much in a single rule, so I feel as if the best path forward is most likely splitting the multi-line rule and auto fix out into its own rule.

**Comment by shashachu** [comment]
Should be fixed by #881
